.github/workflows/integration_tests.yml

```diff
@@ -87,9 +99,16 @@ jobs:
           pip install -r scripts/gha/requirements.txt
           python scripts/gha/restore_secrets.py --passphrase "${{ secrets.TEST_SECRET }}"
           
-      - name: Build and run integration tests
+      - name: Build integration tests (and run Desktop tests)
         # The set up script for Android will download the NDK here.
         env:
           NDK_ROOT: '/tmp/android-ndk-r16b'
         run: |
           python scripts/gha/build_testapps.py --t ${{ github.event.inputs.apis }} --p ${{ matrix.target_platform }} --output_directory ${{ github.workspace }} --use_vcpkg --execute_desktop_testapp --noadd_timestamp
+      - name: Install Cloud SDK for mobile integration tests
+        if: matrix.target_platform != 'Desktop' && ( success() || failure() )
```

Comment:
What's the ( success() || failure() ) about?

Reply:
Those are recognized by GHA. 

success() will be true if there were no failed steps prior to this step (all steps implicitly use success() by default), failure() will be true if any prior step failed. This ensures the tests will run on any integration tests that built successfully, even if others failed (which causes the build step to report failure), instead of the tests being aborted.

There's also an always(), but that includes cancelled(), which we don't want.

Reply:
Would !cancelled() work?  Seems like it would provide more information about the intention of the check and not confused ignorant people like me who think that things either succeed or fail.

Reply:
It would, yes. The behaviour could diverge in the future, and I'm not sure which is more desirable: "not cancelled" would pick up new statuses that Github might add, which could be good or bad. But given the uncertainty, might as go with the more readable option.